arangodb-http/src/test/scala/io/funkode/arangodb/http/ArangoClientHttpSpec.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 Carlos Verdes
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+package io.funkode.arangodb.http
+
+import zio.*
+import zio.http.{Body, Request, Response, TestClient}
+import zio.http.model.*
+import zio.json.{DeriveJsonCodec, JsonCodec}
+import zio.test.*
+import zio.test.Assertion.*
+import io.funkode.arangodb.{AIO, ArangoConfiguration}
+import io.funkode.arangodb.http.JsonCodecs.{given_JsonCodec_CollectionName, given_JsonCodec_CollectionStatus, given_JsonCodec_CollectionType}
+import io.funkode.arangodb.model.*
+
+object ArangoClientHttpSpec extends ZIOSpecDefault:
+
+  given JsonCodec[CollectionInfo] = DeriveJsonCodec.gen[CollectionInfo]
+
+  override def spec: Spec[TestEnvironment, Any] =
+    suite("Arango http client should")(
+      test("Correctly handle empty response from the server instead of the expected one") {
+        val response: PartialFunction[Request, ZIO[Any, Throwable, Response]] = { case _ =>
+          ZIO.succeed(Response.ok)
+        }
+        val assertion = assertError(
+          ArangoError(
+            500,
+            true,
+            s"Empty body in Arango response",
+            -1
+          )
+        )
+        errorTestCase(response, assertion)
+      },
+      test("Correctly handle unexpected error response from the server instead of the expected one") {
+        val response: PartialFunction[Request, ZIO[Any, Throwable, Response]] = {
+          case _ =>
+            ZIO.succeed(Response(Status.Conflict, body = Body.fromString("{\"error\": true}")))
+        }
+        val assertion = assertError(
+          ArangoError(
+            500,
```

Comment:
this test is not correct, we should expect 409 here

Reply:
Fixed